assets/js/block-variation.js

```diff
@@ -0,0 +1,34 @@
+wp.blocks.registerBlockVariation( 'core/query', {
+	name: 'twentytwentyfour/featured-image-query',
+	title: 'Featured Images',
+	description: 'List of posts with featured images',
```

Comment:
Are these translation ready?

Reply:
Looking at https://developer.wordpress.org/block-editor/reference-guides/block-api/block-variations/ and googling some code examples, you want this to be `__( 'List of posts with featured images', 'twentytwentyfour' )`